### ISSUE TITLE
Implement the cumulative cycler

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#2969]
Blocked by #2954

Skeleton structure introduced [here](https://github.com/open-telemetry/opentelemetry-go/pull/2954/files#diff-2006d9fc6ffc5e160dcd724d4fee7ae0be23b124f6078afd75dd701371bfa481R47-R69)

- [ ] Add functionality to fold one aggregation into another so new aggregation periods can be cumulatively folded into exiting ones
- [ ] Implement an algorithm that flushes an aggregator produces the cumulative aggregations for all seen.
- [ ] Full test coverage
- [ ] Benchmarks for CPU and memory performance
